applications/FluidDynamicsApplication/python_scripts/apply_two_fluids_inlet_process.py

```diff
@@ -0,0 +1,135 @@
+import KratosMultiphysics
+import KratosMultiphysics.FluidDynamicsApplication as KratosFluid
+import math
+
+def Factory(settings, Model):
+    if(type(settings) != KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return ApplyTwoFluidsInletProcess(Model, settings["Parameters"])
+
+
+class ApplyTwoFluidsInletProcess(KratosMultiphysics.Process):
+    def __init__(self, Model, settings):
+        KratosMultiphysics.Process.__init__(self)
+
+        # settings for inlet with interface between fluids and separate velocities
+        default_settings = KratosMultiphysics.Parameters("""
+        {
+            "mesh_id"                   : 0,
+            "model_part_name"           : "",
+            "variable_name"             : "VELOCITY",
+            "constrained"               : true,
+            "direction"                 : [1.0,0.0,0.0],
+            "interval"                  : [0.0,"End"],
+            "two_fluid_settings" : {
+                "modulus_air"               : 0.0,
+                "modulus_water"             : 0.0,
+                "interface_normal"          : [0.0,1.0,0.0],
+                "point_on_interface"        : [0.0,0.25,0.0],
+                "inlet_transition_radius"   : 0.05
+            }
+        }
+        """)
+
+        # Trick: allow "modulus" and "direction" to be a double or a string value (otherwise the ValidateAndAssignDefaults might fail).
+        if (settings.Has("modulus_air")):
+            if (settings["two_fluid_settings"]["modulus_air"].IsString()):
+                default_settings["two_fluid_settings"]["modulus_air"].SetString("0.0")
+
+        if (settings.Has("modulus_water")):
+            if (settings["two_fluid_settings"]["modulus_water"].IsString()):
+                default_settings["two_fluid_settings"]["modulus_water"].SetString("0.0")
+
+        if (settings.Has("direction")):
+            if (settings["direction"].IsString()):
+                default_settings["direction"].SetString("automatic_inwards_normal")
+
+        # compare against the appropriate default settings (one or two fluids)
+        settings.ValidateAndAssignDefaults(default_settings)
+
+        # Check input data
+        if (settings["model_part_name"].GetString() == ""):
+            raise Exception("Empty inlet model part name string. Set a valid model part name.")
+        elif (settings["variable_name"].GetString() != "VELOCITY"):
```

Comment:
if it can be only VELOCITY why do you expose it as a parameter at all?

Reply:
The main reason here was to stay as close to the implementation of the "standard" inlet in `apply_inlet_process.py` as possible. In that file, the default settings are defined as follows:

`
        default_settings = KratosMultiphysics.Parameters("""
        {
            "mesh_id"         : 0,
            "model_part_name" : "",
            "variable_name"   : "VELOCITY",
            "modulus"         : 0.0,
            "constrained"     : true,
            "direction"       : [1.0,0.0,0.0],
            "interval"        : [0.0,"End"]
        }
        """)`